source/funkin/util/FileUtil.hx

```diff
@@ -522,6 +522,90 @@ class FileUtil
     #end
   }
 
+  /**
+   * Moves a file from one location to another.
+   * Only works on desktop.
+   *
+   * @param path The path to the file.
+   * @param destination The path to move the file to.
+   */
+  public static function moveFile(path:String, destination:String):Void
+  {
+    #if sys
+    if (doesFileExist(path))
+    {
+      if (isDirectory(path))
+      {
+        trace('WARNING: Path is a directory: $path - attempting to move as a directory instead');
+        moveDir(path, destination);
+        return;
+      }
+
+      var destinationFolder:String = destination;
+      if (Path.extension(destination) != '')
+      {
+        destinationFolder = Path.directory(destination);
+      }
+
+      createDirIfNotExists(destinationFolder);
+      sys.FileSystem.rename(path, Path.join([destinationFolder, Path.withoutDirectory(path)]));
+    }
+    else
+    {
+      throw 'File does not exist: $path';
+    }
+    #else
+    throw 'Direct file moving by path not supported on this platform.';
+    #end
+  }
+
+  /**
+   * Delete a file at the given path.
+   * Only works on desktop.
+   *
+   * @param path The path to the file.
+   */
+  public static function deleteFile(path:String):Void
+  {
+    #if sys
+    sys.FileSystem.deleteFile(path);
+    #else
+    throw 'Direct file deletion by path not supported on this platform.';
+    #end
+  }
+
+  /**
+   * Get a file's size in bytes. Max representable size is ~2.147 GB.
+   * Only works on desktop.
+   *
+   * @param path The path to the file.
+   * @return The size of the file in bytes.
+   */
+  public static function getFileSize(path:String):Int
```

Comment:
Couldn't we just return a `Float` which is a 64-bit floating point number, which leaves us with much bigger values than 2.147 GB